mne/channels/channels.py

```diff
@@ -336,6 +339,68 @@ def rename_channels(info, mapping):
     # it doesn't work)
     info['ch_names'] = [c['ch_name'] for c in chs]
 
+    # assert channel names are unique
+    assert len(info['ch_names']) == np.unique(info['ch_names']).shape[0]
+
+
+def define_sensor(info, mapping):
```

Comment:
I don't like the fact that we start using the word "sensor" while we use channels everywhere. Also as it defines the type it should appear in function name. How about

set_channels_type ?